### PR TITLE
 #12: Add CRUD Operations for project

### PR DESCRIPTION
**Pull request by SakshiShreya**

### CODE REVIEW

File: src/schema/project/project.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      description: description.desc,
+    },
+    startDate: {
+      type: new GraphQLNonNull(DateType),
+      description: description.startDate,
+    },
+    duration: {
+      type: new GraphQLNonNull(DurationInputType),
+      description: description.duration,
+    },
```
Comment (krishna1212031) [inline src/schema/project/project.js]: Instead of duration, we need to store the end date

File: src/schema/project/project.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    status: {
```
Comment (krishna1212031) [inline src/schema/project/project.js:25]: if the project status is not changed on the date of start date. the system will consider it as delayed while making the query. 
start and end date are two types. 
1. scheduled (while creating the project)
2. actual (when the project starts)

File: src/models/projectModel.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  duration: {
+    type: durationSchema,
+    required: [true, "A project must have a duration"],
+  },
+  status: {
+    type: String,
+    required: [true, "A project must have a status"],
+    enum: ["scheduled", "inProgress", "delayed", "completed", "onHold"],
+    default: "scheduled",
+  },
```
Comment (krishna1212031) [inline src/models/projectModel.js:32]: as discussed do needful for status.

**Review by krishna1212031 — CHANGES_REQUESTED** [review]

File: src/schema/project/project.js (hunk 1 above)
Comment (SakshiShreya) [reply src/schema/project/project.js]: @krishna1212031 Removed duration. End date we discussed will be calculated depending on project. So, I am not adding it right now.

File: src/models/projectModel.js (hunk 3 above)
Comment (SakshiShreya) [reply src/models/projectModel.js:32]: @krishna1212031 changed status messages as discussed. I will do the calculation of status while making the GET api.

File: src/schema/project/project.js (hunk 2 above)
Comment (SakshiShreya) [reply src/schema/project/project.js:25]: @krishna1212031 Added the new type of dates as discuessed

**Comment by SakshiShreya** [comment]
@krishna1212031 Can you please review this code again.

**Review by krishna1212031 — APPROVED** [review]

**Comment by SakshiShreya** [comment]
[BE- Project apis](https://trello.com/c/LdP64kuo/9-be-project-apis)

**Pull request merged**